Add sample distribution

AntoniaChroni opened the pull request:
This PR is the updated Rmd script for the sample_distribution_analysis for the PBTA cohort (based on suggestions from the pbta-tumor-evolution-old repo).

Please review the script logic and output file for clarity.

Commit:
add sample distribution

add sample distribution script and files v2

Commit:
add sample distribution

add sample distribution script and files v2

Commit:
add sample distribution updates v3

Commit:
add sample distribution updates v3

jharenza commented on analyses/01_sample_distribution_analysis/01-tumor-descriptor-and-assay-count-plot.Rmd:
```diff
@@ -0,0 +1,285 @@
+          !is.na(pathology_diagnosis),
+          !composition %in% c("Derived Cell Line", "PDX")) %>% 
+         # !is.na(cancer_group) # Discard NAs in cancer type
+  mutate(tumor_descriptor = ifelse(sample_id == "C4106355-Tumor" & tumor_descriptor == "Not Reported", "Recurrence", 
+                                   ifelse(sample_id == "C4113612-Tumor" & tumor_descriptor == "Initial CNS Tumor", "Recurrence", 
+                                          tumor_descriptor)))
```
```suggestion
  mutate(tumor_descriptor = ifelse(sample_id == "C4106355-Tumor", "Recurrence", tumor_descriptor,
                                   ifelse(sample_id == "C4113612-Tumor", "Initial CNS Tumor", tumor_descriptor)))
```

The second ifelse should be if "Not Reported" then "Initial CNS Tumor", but I think it is easier to just remove the if not reported part bc we know what they should be regardless..

jharenza commented on analyses/01_sample_distribution_analysis/01-tumor-descriptor-and-assay-count-plot.Rmd:
```diff
@@ -0,0 +1,285 @@
+# manually annotate them here and these will be updated in the v13 - then discard that part of the code
+
+
+data$tumor_descriptor <- ifelse(data$tumor_descriptor=="Progressive Disease Post-Mortem", "Deceased",
+                                 ifelse(data$tumor_descriptor=="Initial CNS Tumor", "Diagnosis",
+                                 ifelse(data$tumor_descriptor=="Primary Tumor", "Diagnosis", data$tumor_descriptor)))
```
I think it would be easier if you add this logic to the above mutate function - to keep consistency with using tidyverse above over base here, and then you can also combine the two instances of making the tumor "Diagnosis" (ifelse tumor_descriptor %in% c("Initial CNS Tumor", "Primary Tumor")....

jharenza commented:
@AntoniaChroni can you add a gitignore which will remove (ignore) these .DS_store files? They are just hidden computer files we don't want in the repo

Commit:
ignore .DS_Store

# Desktop Services Store - invisible file that gets automatically created anytime you open a folder with Finder on the macOS
.DS_Store

Commit:
ignore .DS_Store

# Desktop Services Store - invisible file that gets automatically created anytime you open a folder with Finder on the macOS
.DS_Store

Commit:
Revert "Revert "ignore .DS_Store""

This reverts commit 88015e3c79ebc67b489da7ce9d234700f3ed31c3.

Commit:
Revert "ignore .DS_Store"

This reverts commit 74ee05e7d891bacc91a6992be520661f7bb00566.

AntoniaChroni committed:
Update analyses/01_sample_distribution_analysis/01-tumor-descriptor-and-assay-count-plot.Rmd

Co-authored-by: Jo Lynne Rokita <jharenza@gmail.com>

Commit:
add sample distribution updates v4

add sample distribution updates v4

AntoniaChroni committed:
Update .gitignore

Commit:
add sample distribution updates v4

add sample distribution updates v4
minor changes: mutate(tumor_descriptor...) and changed the status in tumor_descriptor of 2 samples (C4106355-Tumor, C4113612-Tumor)

Commit:
Auto stash before rebase of "Update-.gitignore-with-.DS_Store"

Commit:
add sample distribution updates v5

update sample distribution script:
- not saved plots
- change layout
- added script to answer with more specific questions about paired assays, #patients, cancer_types
- added low survival cohorts

AntoniaChroni commented:
@jharenza I have updated the sample distribution script:
- not saved plots
- change layout
- added script to answer with more specific questions about paired assays, #patients, cancer_types
- added low survival cohorts

jharenza commented on analyses/01_sample_distribution_analysis/01-tumor-descriptor-and-assay-count-plot.Rmd:
```diff
@@ -0,0 +1,504 @@
+genomic_df <- tumor_descriptor_df %>%
+  filter(experimental_strategy %in% c("WGS", "WXS")) %>%
+  group_by(Kids_First_Participant_ID) %>%
+  summarize(descriptors = paste(sort(unique(tumor_descriptor)),  
+                                collapse = ", "),
+            experimental_strategy = paste(sort(unique(experimental_strategy)),
+                                          collapse = ", "),
+            cancer_group = paste(sort(unique(cancer_group)),
+                                          collapse = ", "))
```
want to get rid of unique here to make sure you see _all_ of the samples available.

jharenza replied:
group_by(Kids_First_Participant_ID, match_id, cancer_group) %>%

Commit:
staged .DS_Store file

Commit:
Updates on add-sample-distribution analysis

- Updated the script
- Updated files in the "Results" folder

AntoniaChroni commented:
I made updates on the add-sample-distribution.

Issues so far:
- Join function for original df and filtered df results to additional samples in the table. I did a workaround for this, but I don't know why this happens.
- Paired assays results to no pairs between any combination of different time of points. This doesn't seem right.

Commit:
added README.md file

added README.md file

Commit:
updates on the script and files

Issues so far:

Join function for original df and filtered df results to additional samples in the table. I did a workaround for this, but I don't know why this happens.
Paired assays results to no pairs between any combination of different time of points. This doesn't seem right.

Commit:
Updates on script, results, readme file

I made some edits to the script that help with identifying pairs of assays and pairs of time points across patient samples.

Commit:
Revert "Updates on script, results, readme file"

This reverts commit b943e83b8882be0d17fab79a135433fc6156f41d.

Commit:
Updates on script, results, readme file

I made some edits to the script that help with identifying pairs of assays and pairs of time points across patient samples.

Commit:
Updates on script, results, readme file

I made some edits to the script that help with identifying pairs of assays and pairs of time points across patient samples.

Commit:
Updates on script, results, readme file

I made some edits to the script that help with identifying pairs of assays and pairs of time points across patient samples.

rjcorb commented on analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count-plot.Rmd:
```diff
@@ -0,0 +1,534 @@
+
+
+# File path to input directory
+input_dir <-
```
```suggestion
analysis_dir <- file.path(root_dir, "analyses", "sample-distribution-analysis")
input_dir <- file.path(analysis_dir, "input")
```

AntoniaChroni replied:
This doesn't work on my end for some reason. I left the directories as they are for now.

rjcorb commented on analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count-plot.Rmd:
```diff
@@ -0,0 +1,534 @@
+
+# File path to results directory
+results_dir <-
+  file.path(root_dir, "results")
```
```suggestion
  file.path(analysis_dir, "results")
```

rjcorb commented on analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count-plot.Rmd:
```diff
@@ -0,0 +1,534 @@
+# THIS PART I AM NOT SURE OF 
+paired_assays <- assays %>%
+  group_by(Kids_First_Participant_ID, match_id) %>%
+  summarise(experimental_strategy = str_c(experimental_strategy, collapse = ";")) %>%
```
I think the issue here is that `match_id` isn't a timepoints-specific identifier, since `experimental_strategy` is included in the ID. Maybe this isn't necessary and can be removed? Then you should get the desired summary by patient and time point

rjcorb replied:
This is the output I received when removing `experimental_strategy` from `match_id`: 

```
# A tibble: 6 × 4
# Groups:   Kids_First_Participant_ID [5]
  Kids_First_Participant_ID match_id                                                                         experimental_strategy experimental_strategy_sum
  <chr>                     <chr>                                                                            <chr>                 <chr>                    
1 PT_00G007DM               CNS Embryonal tumor_7316-2577_Solid Tissue_Recurrence                            RNA-Seq;WGS           DNA                      
2 PT_00G007DM               Embryonal tumor with multilayer rosettes_7316-272_Solid Tissue_Initial CNS Tumor RNA-Seq;WGS           DNA                      
3 PT_017WC8PS               CNS Embryonal tumor_C4114227-Tumor_Solid Tissue_Initial CNS Tumor                RNA-Seq;WXS           DNA                      
4 PT_01HNFSBZ               High-grade glioma_C4115334-Tumor_Solid Tissue_Initial CNS Tumor                  RNA-Seq;WXS           DNA                      
5 PT_01MZ62KG               Pineoblastoma_7316-447_Solid Tissue_Progressive                                  RNA-Seq;WGS           DNA                      
6 PT_01SH4F1X               Ganglioglioma_7316-6365_Solid Tissue_Initial CNS Tumor                           RNA-Seq;WGS           DNA        
```

rjcorb commented on analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count-plot.Rmd:
```diff
@@ -0,0 +1,534 @@
+         !composition %in% c("Derived Cell Line", "PDX"),
+         !experimental_strategy %in% c("Methylation", "Targeted Sequencing")) %>%
+           # !is.na(cancer_group) # Discard NAs in cancer type
+  mutate(match_id = paste(cancer_group, sample_id, composition, experimental_strategy, tumor_descriptor, sep = "_")) %>% 
```
```suggestion
  mutate(match_id = paste(cancer_group, sample_id, composition,  tumor_descriptor, sep = "_")) %>% 
```

rjcorb commented on analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count-plot.Rmd:
```diff
@@ -0,0 +1,534 @@
+paired_assays <- assays %>%
+  group_by(Kids_First_Participant_ID, match_id) %>%
+  summarise(experimental_strategy = str_c(experimental_strategy, collapse = ";")) %>%
+  mutate(experimental_strategy_sum = case_when(grepl("WGS", experimental_strategy) ~ "DNA",
+                                                grepl("WXS", experimental_strategy) ~ "DNA", 
+                                                grepl("RNA-Seq", experimental_strategy) ~ "RNA",
+                                                TRUE ~ experimental_strategy)) 
```
Is `experimental_strategy_sum` supposed to be indicating whether a sample has RNA, DNA, or both? If so, this logic should work: 
```suggestion
  mutate(experimental_strategy_sum = case_when(grepl("RNA-Seq", experimental_strategy) & grepl("WGS|WXS", experimental_strategy) ~ "DNA,RNA",
                                                grepl("WXS|WGS", experimental_strategy) ~ "DNA", 
                                                grepl("RNA-Seq", experimental_strategy) ~ "RNA"
                                                )) 
```

rjcorb commented on analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count-plot.Rmd:
```diff
@@ -0,0 +1,534 @@
+assays <- pbta %>%
+  select(Kids_First_Participant_ID, tumor_descriptor_final, match_id, experimental_strategy, cancer_group) %>%
+  unique() %>%
+  arrange(Kids_First_Participant_ID)
```
```suggestion
  arrange(Kids_First_Participant_ID, experimental_strategy))
```

arranging by `experimental_strategy` will ensure ordering of assays is the same when summarizing below

rjcorb commented on analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count-plot.Rmd:
```diff
@@ -0,0 +1,534 @@
+The following table was supposed to provide matched different time points and summarize the information under the "tumor_descriptor_sum" column.
+But it doesn't work for now.
+
+Also, notice how for the first patient with PT_00G007DM, there are two different cancer_group assigned, which creates double events per assay.
```
I think the suggestions above should address some of these issues. I think it's ok if a patient has time points with different cancer groups, but maybe @jharenza  can confirm.

AntoniaChroni replied:
It looks like patients with multiple time points might have different diagnosis at different stage of the disease. 
@jharenza may confirm.

rjcorb commented on analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count-plot.Rmd:
```diff
@@ -0,0 +1,534 @@
+    tumor_descriptor=="Diagnosis" & tumor_descriptor=="Metastatic" ~ "Diagnosis-Metastatic",
+    tumor_descriptor=="Diagnosis" & tumor_descriptor=="Unvailable" ~ "Diagnosis-Unvailable",
+    TRUE ~ "Unmatched"
+  ))
```
I think this produces the desired output: 

```
tumor_descriptor_df_Diagnosis <- tumor_descriptor_df %>%
  group_by(Kids_First_Participant_ID) %>%
  summarise(tumor_descriptor_sum = str_c(tumor_descriptor, collapse = "-")) %>%
  mutate(tumor_descriptor_sum = case_when(
    grepl("-", tumor_descriptor_sum) ~ tumor_descriptor_sum,
    TRUE ~ "Unmatched"
  ))
```

rjcorb commented on analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count-plot.Rmd:
```diff
@@ -0,0 +1,534 @@
+
+
+tumor_descriptor_df_Diagnosis %>%
+  regulartable() %>%
```
```suggestion
  group_by(tumor_descriptor_sum) %>%
  tally() %>%
  arrange(desc(n)) %>% 
  regulartable() %>%
```

rjcorb commented on analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count-plot.Rmd:
````diff
@@ -0,0 +1,534 @@
+
+```{r}
+genomic_df <- tumor_descriptor_df %>%
+  filter(!(experimental_strategy_sum == "RNA")) %>% 
````
```suggestion
   filter(!grepl("RNA", experimental_strategy_sum)) %>%
```

AntoniaChroni replied:
Here, we want to exclude samples with RNA-seq assay only. If samples have both assays, then we will keep them.
The suggested code will remove the samples with RNA paired with DNA as well. Is this what you wanted to suggest here?

AntoniaChroni replied:
Here, we want to exclude samples with RNA-seq assay only. If samples have both assays, then we will keep them.
The suggested code will remove the samples with RNA paired with DNA as well. Is this what you wanted to suggest here?

rjcorb replied:
I see, in that case it is fine as is!

rjcorb commented on analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count-plot.Rmd:
````diff
@@ -0,0 +1,534 @@
+
+```{r}
+transcriptomic_df <- tumor_descriptor_df %>%
+  filter(experimental_strategy == "RNA-Seq") %>%
````
```suggestion
  filter(grepl("RNA-Seq", experimental_strategy)) %>%
```

rjcorb commented on analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count-plot.Rmd:
```diff
@@ -0,0 +1,534 @@
+# if we perform a full join here using the Kids_First_Participant_ID and descriptors, we 
+# will get NAs in columns where pairs don't exist and can use this information 
+# to count
+paired_df <-
+  full_join(genomic_df, transcriptomic_df,
+                   by = c("cancer_group", "Kids_First_Participant_ID", "descriptors", "experimental_strategy"))
+
```
I think this is something you can obtain from the `paired_assays` df, once it's modified. Unless there's some difference here that I'm not seeing.

AntoniaChroni replied:
This was left from a previous version. I fixed it now.

rjcorb reviewed (changes requested):
I think my suggestions will resolve some of the joining issues you're having. Let me know if you have any questions!

AntoniaChroni committed:
Update analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count-plot.Rmd

Co-authored-by: Ryan Corbett <corbettr@chop.edu>

AntoniaChroni committed:
Update analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count-plot.Rmd

Co-authored-by: Ryan Corbett <corbettr@chop.edu>

AntoniaChroni committed:
Update analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count-plot.Rmd

Co-authored-by: Ryan Corbett <corbettr@chop.edu>

AntoniaChroni committed:
Update analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count-plot.Rmd

Co-authored-by: Ryan Corbett <corbettr@chop.edu>

AntoniaChroni committed:
Update analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count-plot.Rmd

Co-authored-by: Ryan Corbett <corbettr@chop.edu>

Commit:
Updates on add-sample-distribution after RC Review

- Updated script
- Updates README.md file
- Updated results folder

Commit:
Merge remote-tracking branch 'origin/add-sample-distribution' into add-sample-distribution

Commit:
updates

updates on the script

Commit:
minor changes in the script

- Updated script
- Updated results folder

rjcorb commented on analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count-plot.Rmd:
```diff
@@ -0,0 +1,519 @@
+
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+setwd(root_dir)
+
```
```suggestion
analysis_dir <- file.path(root_dir, "analyses", "sample-distribution-analysis")
```

rjcorb commented on analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count-plot.Rmd:
```diff
@@ -0,0 +1,519 @@
+
+
+# File path to input directory
+input_dir <- file.path(root_dir, "input")
```
```suggestion
input_dir <- file.path(analysis_dir, "input")
```

rjcorb commented on analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count-plot.Rmd:
````diff
@@ -0,0 +1,519 @@
+```{r}
+tumor_descriptor_df_Diagnosis <- tumor_descriptor_df %>%
+  group_by(Kids_First_Participant_ID) %>%
+  summarise(tumor_descriptor_sum = str_c(tumor_descriptor, collapse = "-")) %>%
````
```suggestion
  arrange(tumor_descriptor) %>%
  summarise(tumor_descriptor_sum = str_c(tumor_descriptor, collapse = "-")) %>%
```

arranging by `tumor_descriptor` to make ordering consistent when grouping

rjcorb reviewed (changes requested):
I made a few additional inline comments. I think an `analysis_dir` is needed to define the `sample-distribution-analysis` directory path. Right now the `input_dir` and `results_dir` paths are set to `root_dir/input` and `root_dir/results`, respectively, which are incorrect.

AntoniaChroni committed:
Update analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count-plot.Rmd

Co-authored-by: Ryan Corbett <corbettr@chop.edu>

AntoniaChroni committed:
Update analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count-plot.Rmd

Co-authored-by: Ryan Corbett <corbettr@chop.edu>

AntoniaChroni committed:
Update analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count-plot.Rmd

Co-authored-by: Ryan Corbett <corbettr@chop.edu>

AntoniaChroni committed:
Update analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count-plot.Rmd

Co-authored-by: Ryan Corbett <corbettr@chop.edu>

Commit:
Updates after PR review

- integrated RC review about analysis_dir etc
- added genomic_df_filter df
- updates on script, README, results

Commit:
Merge remote-tracking branch 'origin/add-sample-distribution' into add-sample-distribution

AntoniaChroni commented:
This is updated based on @rjcorb suggestions. I also added/edited to create a file needed for the next module. It is now ready for review @jharenza 

When identifying multiple time points, there are cases such as Diagnosis, Recurrence, Recurrence (one time point showing multiple times). Any thoughts on why this might be happening?

rjcorb commented on analyses/sample-distribution-analysis/01-tumor-descriptor-and-assay-count-plot.Rmd:
```diff
@@ -0,0 +1,507 @@
+
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+setwd(root_dir)
+<<<<<<< HEAD
+
+=======
+>>>>>>> add-sample-distribution
```
looks like some code annotation got placed here on accident.

AntoniaChroni replied:
thanks for catching this!

rjcorb reviewed (changes requested):
There's a code chunk that needs to be deleted in the R script, otherwise looks good!

Commit:
removed misplaced code annotation
